### PR TITLE
Fixes Issue 1331

### LINKED ISSUE #1331: Issue with reading amber prmtop if num items is divisible with items per line
### Expected behaviour
Was expecting the prmtop to load without issue.

### Actual behaviour
Failed to load with the error:
Error: could not convert string to float: %FLAG ATOMIC_NUM

### Code to reproduce the behaviour

``` python
import MDAnalysis as mda

u = mda.Universe("HPC.prmtop", "03_heat.nc")

```

### Currently version of MDAnalysis:
(0.16.0)

### Further analysis
My code worked with some of my prmtops but not others
I examined the differences between the files and noticed this:

Files that didn't work looked like this:
```
  0.00000000E+00  5.15691090E+00  0.00000000E+00  5.02935480E+00  0.00000000E+00
  0.00000000E+00 -1.24276086E+01  7.61692140E+00 -1.04596002E+01 -8.52803640E+00
%FLAG ATOMIC_NUMBER
%FORMAT(10I8)
       1       8       6       1       8       6       1       6       1       1
       8       6       1       7       1       6       1       8       1       6
```

Files that did work looked like this:
```
  7.59869910E+00 -1.51973982E+01  7.59869910E+00  7.59869910E+00 -1.51973982E+01
  7.59869910E+00  7.59869910E+00
%FLAG ATOMIC_NUMBER
%FORMAT(10I8)
       1       8       6       1       8       6       1       6       1       1
       8       6       1       7       1       6       1       8       1       6
```

It seems that files where the number of items was exactly divisible by number of items per lines then it fails to load the prmtop.

Looking at the relevant code in the source, I noticed this on line 185 of `topology/TOPParser.py`
```python
              except KeyError:
                    next_getter = self.skipper
                else:
                    num = sys_info[sect_num]
                    numlines = (num // per_line) + 1 #<----HERE

                    attrs[name] = func(atoms_per, numlines)
```
Where I have marked HERE you seem to always be rounding up. You should not round up when the remainder of the division is zero.

I amended the code to read:
```python
              except KeyError:
                    next_getter = self.skipper
                else:
                    num = sys_info[sect_num]
                    numlines = (num // per_line)
                    if num % per_line != 0:
                        numlines += 1

                    attrs[name] = func(atoms_per, numlines)
```

It now works as expected regardless of input.

I hope that this makes sense. Please let me know if there is anything else you would like me to clarify.

### PR DESCRIPTION
**Pull request by QuantumEntangledAndy**
This handles the case where number of items is exactly divisible by number of items per line. Previously the code would always round up, where instead it should only round up when not exactly divisible.

Fixes #1331 

Changes made in this Pull Request:
 - Changes how amber prmtops are loaded such that the number of lines is only rounded up when neede
- Added to the change log
- Added a prmtop to the test suite that fails without this corrected rounding

PR Checklist
------------
 - [x] Tests?
 - [ ] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: testsuite/MDAnalysisTests/datafiles.py [hunk 1]
```diff
@@ -90,6 +90,7 @@
     "PRM12", "TRJ12_bz2",  # Amber (v12 format, Issue 100)
     "PRMncdf", "TRJncdf", "NCDF",  # Amber (netcdf)
     "PFncdf_Top", "PFncdf_Trj", # Amber ncdf with Positions and Forces
+    "PRMcs" # Amber (format, Issue 1331)
```
Comment (jbarnoud) [inline testsuite/MDAnalysisTests/datafiles.py]: A comma is missing here causing the tests to fail.

File: package/CHANGELOG [hunk 2]
```diff
@@ -37,6 +37,8 @@ Fixes
   * Fix hbond_analysis cannot deal with Universe where no two atoms are with 3A.
     (PR #1325)
   * Fix PDBParser docs for conect (issue #1246)
+  * Fixed bug where amber topology files would fail to load if number of atoms was
```
Comment (jbarnoud) [inline package/CHANGELOG]: You should add your github handle on line 18 with the other contributors to this version

**Comment by QuantumEntangledAndy** [comment]
Yes a fairly minor change, I wasn't really sure it was worth a PR. But from the discussion in the issue #1333 it seems that I should.

I believe I have corrected all of your issues, sorry about the comma that was embarrassing.

**Comment by jbarnoud** [comment]
Your PR is now merged. Thank you for your contribution.

Don't worry about the comma, this is what the tests and the reviews are for :wink:

**Pull request merged**